repository/src/main/java/org/alfresco/repo/preference/script/ScriptPreferenceService.java

```diff
@@ -76,8 +77,16 @@ public NativeObject getPreferences(String userName)
     public NativeObject getPreferences(String userName, String preferenceFilter)
     {
         // It's a tad unusual to return a NativeObject like this - at least within Alfresco.
-        // But we can't change it to e.g. a ScriptableHashMap as the API is published.
-        Map<String, Serializable> prefs = this.preferenceService.getPreferences(userName, preferenceFilter);        
+        // But we can't change it to e.g. a ScriptableHashMap as the API is published.
+        Map<String, Serializable> prefs = this.transactionService.getRetryingTransactionHelper().doInTransaction(new RetryingTransactionCallback<Map<String, Serializable>>()
```

Comment:
Why do you need to wrap the calls to preferenceService in a transaction? Isn't it better to modify the spring bean to use the upper case "PreferenceService" with a transaction interceptor instead?